app/controllers/markets_controller.rb

```diff
@@ -1,12 +1,58 @@
 class MarketsController < ApplicationController
 
+  def home
+
+  end
+
   def index
+    @is_market = true if session[:user_type] == "market"
+    user = params[:user_type]
+    session[:user_type] = user unless user.nil?
     @markets = Market.all
+
   end
 
   def show
-    id = params[:id]
-    @market = Market.find(id)
+    @market = Market.find(params[:id])
+    @is_market = true if session[:user_type] == "market"
```

Comment:
By using the one-line conditionals here with the `if`s, your code will attempt execution on each individual conditional. The problem with this is that each conditional is actually mutually exclusive, so if the first one is true, it should not continue attempting to execute the rest of the conditionals. 
I think that an if/elsif/else condition would be better suited here.